transport/http/transport.go

```diff
@@ -50,6 +54,16 @@ type TransportOption func(*transportConfig)
 
 func (TransportOption) httpOption() {}
 
+// NilOption is empty option - example use case:
```

Comment:
You can just do a slice of options, that's what I usually do, and then if you don't have the option, your slice is of length 0, ie what @abhinav proposed above.

This is the standard for all other options in yarpc-go.